edunext_openedx_extensions/edunext/middleware.py

```diff
@@ -78,3 +81,68 @@ def clear_cache(sender, instance, **kwargs):  # pylint: disable=unused-argument
         """
         cache_key = "redirect_cache." + fasthash(instance.domain)
         cache.delete(cache_key)  # pylint: disable=maybe-no-member
+
+
+class PathRedirectionMiddleware(object):
+    """
+    Middleware to create custom responses based on the request path
+    """
+    LOGIN_REQUIRED = "login_required"
+    NOT_FOUND = "not_found"
+    NOT_FOUND_LOGGEDIN = "not_found_loggedin"
+    NOT_FOUND_LOGGEDOUT = "not_found_loggedout"
+    REDIRECT_ALWAYS = "redirect_always"
+    REDIRECT_LOGGEDIN = "redirect_loggedin"
+    REDIRECT_LOGGEDOUT = "redirect_loggedout"
+
+    def process_request(self, request):
+        """
+        This middleware processes the path of every request and determines if there
+        is a configured action to take.
+        """
+
+        if microsite.has_override_value("EDNX_CUSTOM_PATH_REDIRECTS"):
+            redirects = microsite.get_value("EDNX_CUSTOM_PATH_REDIRECTS", {})
+
+            for regex, values in redirects.iteritems():
+
+                if isinstance(values, dict):
+                    key = next(iter(values))
+                else:
+                    key = values
+
+                path = request.path_info
+                regex_path_match = re.compile(regex.format(
+                    COURSE_ID_PATTERN=settings.COURSE_ID_PATTERN,
+                    USERNAME_PATTERN=settings.USERNAME_PATTERN,
+                ))
+
+                if regex_path_match.match(path):
+                    if key == self.LOGIN_REQUIRED:
```

Comment:
The list of possbile keys can grow in a way we should worry about using a different strategy than if-cases?